filters/shedder/admission_test.go

```diff
@@ -208,6 +240,127 @@ func TestAdmissionControl(t *testing.T) {
 	}
 }
 
+func TestAdmissionControlChainOnlyBackendErrorPass(t *testing.T) {
+	dm := metrics.Default
+	t.Cleanup(func() { metrics.Default = dm })
+	m := &metricstest.MockMetrics{}
+	metrics.Default = m
+	defer m.Close()
+
+	// backend with 50% error rate
+	var cnt int64
+	backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		i := atomic.AddInt64(&cnt, 1)
+		if i%2 == 0 {
+			w.WriteHeader(http.StatusInternalServerError)
+		} else {
+			w.WriteHeader(http.StatusOK)
+		}
+	}))
+	defer backend.Close()
+
+	spec := NewAdmissionControl(Options{}).(*AdmissionControlSpec)
+
+	argsLeaf := make([]interface{}, 0, 6)
+	argsLeaf = append(argsLeaf, "testmetric-leaf", "active", "5ms", 5, 5, 0.9, 0.95, 1.0)
+	fLeaf, err := spec.CreateFilter(argsLeaf)
```

Comment:
I think filter creation in this tests obscures the main test logic. Do we need it here?

Reply:
Strictly we don't need it but creating invalid filters is easy to do with so many args, so I think it's better to bail out early.